src/Functions/abtesting.cpp

```diff
@@ -0,0 +1,273 @@
+#if !defined(ARCADIA_BUILD)
+
+#include <math.h>
+#include <sstream>
+
+#include <DataTypes/DataTypeString.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnsNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/abtesting.h>
+#include <IO/WriteHelpers.h>
+#include <IO/WriteBufferFromOStream.h>
+
+#define STATS_ENABLE_STDVEC_WRAPPERS
+#include <stats.hpp>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int BAD_ARGUMENTS;
+}
+
+static const String BETA = "beta";
+static const String GAMMA = "gamma";
+
+template <bool higher_is_better>
+Variants bayesian_ab_test(String distribution, PODArray<Float64> & xs, PODArray<Float64> & ys)
+{
+    const size_t r = 1000, c = 100;
+
+    Variants variants(xs.size(), {0.0, 0.0});
+    std::vector<std::vector<Float64>> samples_matrix;
+
+    if (distribution == BETA)
+    {
+        Float64 alpha, beta;
+
+        for (size_t i = 0; i < xs.size(); ++i)
+            if (xs[i] < ys[i])
+                throw Exception("Conversions cannot be larger than trials", ErrorCodes::BAD_ARGUMENTS);
+
+        for (size_t i = 0; i < xs.size(); ++i)
+        {
+            alpha = 1.0 + ys[i];
+            beta = 1.0 + xs[i] - ys[i];
+
+            samples_matrix.emplace_back(stats::rbeta<std::vector<Float64>>(r, c, alpha, beta));
+        }
+    }
+    else if (distribution == GAMMA)
+    {
+        Float64 shape, scale;
+
+        for (size_t i = 0; i < xs.size(); ++i)
+        {
+            shape = 1.0 + xs[i];
+            scale = 250.0 / (1 + 250.0 * ys[i]);
+
+            std::vector<Float64> samples = stats::rgamma<std::vector<Float64>>(r, c, shape, scale);
```

Comment:
I don't know how to replace vector with PODArray because external module `stats::rgamma` doesn't support PODArray. 
So I wrote this code. If there is a way to use PODArray I will modify it.